addons/mail/models/mail_activity.py

```diff
@@ -166,47 +169,75 @@ def _check_access(self, operation):
         doc_operation = 'read' if operation == 'read' else 'write'
         activity_to_documents = dict()
         for activity in self.sudo():
-            activity_to_documents.setdefault(activity.res_model, list()).append(activity.res_id)
+            if activity.res_model and activity.res_id:
+                activity_to_documents.setdefault(activity.res_model, list()).append(activity.res_id)
+            else:
+                if (operation != 'create' and self.env.user.id != activity.user_id.id):
+                    self._raise_access_exception(operation,_("You can only access your own records."))
+                elif operation =='create' and not self.env.user.has_group('base.group_user'):
+                    self._raise_access_exception(operation,_("Only employee can create reminder."))
         for model, res_ids in activity_to_documents.items():
             self.env[model].check_access_rights(doc_operation, raise_exception=True)
             try:
                 self.env[model].browse(res_ids).check_access_rule(doc_operation)
             except exceptions.AccessError:
-                raise exceptions.AccessError(
-                    _('The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n\n(Document type: %s, Operation: %s)') %
-                    (self._description, operation))
+                self._raise_access_exception(operation)
```

Comment:
Is it really necessary to encapsulate the access error into a call ? I think it adds unnecessary code layers.

Reply:
It's mainly to reduce repetitions of code since the error message is used in 4 different places with small differences.